src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -137,6 +141,15 @@ public Channel createChannel(Course course, Channel channel, Optional<User> crea
         return savedChannel;
     }
 
+    public Set<User> registerUsersToChannel(boolean addAllStudents, boolean addAllTutors, boolean addAllInstructors, List<String> usersLoginsToRegister, Course course,
+            Channel channel) {
+        Set<User> usersToRegister = new HashSet<>();
+        usersToRegister.addAll(conversationService.findUsersInDatabase(course, addAllStudents, addAllTutors, addAllInstructors));
+        usersToRegister.addAll(conversationService.findUsersInDatabase(usersLoginsToRegister.stream().toList()));
```

Comment:
This is already a list. Why do you have to create a stream and a list again?

Reply:
It seems that we have overseen this, will remove the unnecessary conversion